### PR TITLE
ADS-963: companion slots which are configured as 'false' f or a given screensize should never render at that screensize even when lazy-loading of the slot is enabled

### PR DESCRIPTION
**Pull request by andrewgeorgiou1981**
@adgad @VladDubrovskis

### CODE REVIEW

**Comment by VladDubrovskis** [comment]
Looks good, just need a test to keep the coverage up.
Possibly try and use stub/spy in for the "fire" and the "hasValidSize"?

File: test/qunit/slots.test.js [hunk 1]
```diff
@@ -886,6 +886,15 @@ QUnit.test('lazy loading a companion slot', function(assert) {
 	this.utils.broadcast('masterLoaded', {}, node);
 });
 
+QUnit.test('companion slots which are configured as false for a specific screensize should not render at that screensize', function(assert) {
+	const slotHTML = '<div data-o-ads-name="lazy-companion-test" data-o-ads-lazy-load="true" data-o-ads-formats-default="false" data-o-ads-formats-large="false" data-o-ads-formats-small="false" data-o-ads-formats-medium="false" data-o-ads-formats-extra="false" style="position: absolute; left: -1000px; top: -1000px"></div>';
+	const node = this.fixturesContainer.add(slotHTML);
+	this.ads.init();
+	const slot = this.ads.slots.initSlot(node);
+	const renderSpy = this.spy(slot, 'render');
```
Comment (VladDubrovskis) [inline test/qunit/slots.test.js]: add renderSpy.restore() to and of test, which will remove the spy from the call, as might break other tests

**Comment by adgad** [comment]
LGTM

**Comment by VladDubrovskis** [comment]
👍 LGTM just add the restore call as per other comment

**Pull request merged**